codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGeneration.kt

```diff
@@ -0,0 +1,281 @@
+        "StdError" to RuntimeType.StdError
+    )
+    private val operationRegistryName = "${service.getContextualName(service)}OperationRegistry"
+    private val operationRegistryBuilderName = "${operationRegistryName}Builder"
+    private val genericArguments = "B, " + operations.mapIndexed { i, _ -> "Op$i, In$i" }.joinToString()
+    private val operationRegistryNameWithArguments = "$operationRegistryName<$genericArguments>"
+    private val operationRegistryBuilderNameWithArguments = "$operationRegistryBuilderName<$genericArguments>"
+
+    fun render(writer: RustWriter) {
+        renderOperationRegistryStruct(writer)
+        renderOperationRegistryBuilderStruct(writer)
+        renderOperationRegistryBuilderError(writer)
+        renderOperationRegistryBuilderDefault(writer)
+        renderOperationRegistryBuilderImpl(writer)
+        renderRouterImplFromOperationRegistryBuilder(writer)
+    }
+
+    /*
+     * Renders the OperationRegistry structure, holding all the operations and their generic inputs.
+     */
+    private fun renderOperationRegistryStruct(writer: RustWriter) {
+        // A lot of things can become pretty complex in this type as it will hold 2 generics per operation
+        Attribute.Custom("allow(clippy::all)").render(writer)
+        writer.rustBlock(
+            """
+            pub struct $operationRegistryNameWithArguments
+            """.trimIndent()
+        ) {
+            val members = operationNames
+                .mapIndexed { i, operationName -> "$operationName: Op$i" }
+                .joinToString(separator = ",\n")
+            rustTemplate(
+                """
+                $members,
+                _phantom: #{Phantom}<(B, ${phantomMembers()})>,
+                """,
+                *codegenScope
+            )
+        }
+    }
+
+    /*
+     * Renders the OperationRegistryBuilder structure, used to build the OperationRegistry and then convert it
+     * into a Smithy Router.
+     */
+    private fun renderOperationRegistryBuilderStruct(writer: RustWriter) {
+        // A lot of things can become pretty complex in this type as it will hold 2 generics per operation
+        Attribute.Custom("allow(clippy::all)").render(writer)
+        writer.rustBlock(
+            """
+            pub struct $operationRegistryBuilderNameWithArguments
+            """.trimIndent()
+        ) {
+            val members = operationNames
+                .mapIndexed { i, operationName -> "$operationName: Option<Op$i>" }
+                .joinToString(separator = ",\n")
+            rustTemplate(
+                """
+                $members,
+                _phantom: #{Phantom}<(B, ${phantomMembers()})>,
+                """,
+                *codegenScope
+            )
+        }
+    }
+
+    /*
+     * Renders the OperationRegistryBuilder Error type, used to error out in case there are uninitialized fields.
+     * This structs implement Debug, Display and std::error::Error.
```

Comment:
```suggestion
     * Renders the `OperationRegistryBuilderError` type, used to error out in case there are uninitialized fields.
     * This enum implement `Debug`, `Display` and `std::error::Error`.
```